### DISCUSSION TITLE
Questions about keymap when I use vscode-neovim and lazyvim extras.

### DISCUSSION THREAD

**Discussion by zwyyy456** [LazyVim/LazyVim#2041] [Q&A]
I have enabled the vscode lazyvim extras. And I use vscode-neovim on vscode.
My `init.lua` is as follow.

```lua
require("config.lazy")
if vim.g.vscode then
else
    require("config.osc52")
end
```

Most things work well. But there still some problems about keymaps. In the past, I disable  lazyvim when using vscode and only lazy.nvim to install some plugins.  I often use `<C-w>(h/j/k/l)` to navigate vscode panel.

However, when I enable lazyvim and vscode lazyvim extras. `<C-w>(h/j/k/l)` doesn't work in vscode. Can anyone tell meWhich neovim plugin occupies the keymap, since there are only 11 plugins are enabled.

**Comment by abeldekat** [comment]
`<C-w>(h/j/k/l)`: builtin vim window navigation. I don't think any plugin overrides those keys.

`LazyVim` defines [convenience](https://github.com/LazyVim/LazyVim/blob/68ff818a5bb7549f90b05e412b76fe448f605ffb/lua/lazyvim/config/keymaps.lua#L15) mappings.

However, when in vscode, your config does not load `LazyVim`. Therefore, those `keymappings` are not loaded.

`your init.lua`:

```
-- require("config.lazy")
if vim.g.vscode then
  -- if true then
  -- Vscoe extension
  require("vscode.config.options")
  require("vscode.config.keymaps")
  require("vscode.config.lazy")
else
  -- bootstrap lazy.nvim, LazyVim and your plugins
  require("config.lazy")
  require("config.osc52")
end
```

You might want to consider an approach more closely to `LazyVim starter`. Add [`extra.vscode`](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/extras/vscode.lua) and override that module in your own `extra.vscode`. In that file you can tailor the config for vscode, using for example the `cond` function to enable/disable plugins.
